core_impl/src/test/java/com/google/instrumentation/trace/TimestampConverterTest.java

```diff
@@ -16,18 +16,23 @@
 import static com.google.common.truth.Truth.assertThat;
 
 import com.google.instrumentation.common.Timestamp;
+import com.google.instrumentation.internal.TestClock;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
 /** Unit tests for {@link TimestampConverter}. */
 @RunWith(JUnit4.class)
 public class TimestampConverterTest {
+  private final Timestamp timestamp = Timestamp.create(1234, 5678);
+  private final TestClock testClock = TestClock.create(timestamp);
+
   @Test
   public void convertNanoTime() {
-    Timestamp timestamp = Timestamp.create(1234, 5678);
-    TimestampConverter timeConverter = new TimestampConverter(timestamp, 2345);
-    assertThat(timeConverter.convertNanoTime(1234)).isEqualTo(timestamp.addNanos(-1111));
-    assertThat(timeConverter.convertNanoTime(3456)).isEqualTo(timestamp.addNanos(1111));
+    TimestampConverter timeConverter = TimestampConverter.now(testClock);
+    assertThat(timeConverter.convertNanoTime(testClock.nowNanos() - 1234))
+        .isEqualTo(timestamp.addNanos(-1234));
+    assertThat(timeConverter.convertNanoTime(testClock.nowNanos() + 1234))
+        .isEqualTo(timestamp.addNanos(1234));
```

Comment:
I just had an idea for making the test more similar to the expected use case, by mocking Clock and giving it low precision.  I think the existing test is fine, though.  Here are the steps:

Set the mock's result for its two methods like this:
now(): TimeStamp.create(1234, 0)
nowNanos(): 100

Create the TimestampConverter, and then advance the clock to:
now(): TimeStamp.create(1234, 0)
nowNanos(): 300

Call convertNanoTime with 300, and check that the time is TimeStamp.create(1234, 200).